boefjes/boefjes/api.py

```diff
@@ -0,0 +1,182 @@
+import base64
+import logging
+import multiprocessing
+from datetime import datetime, timezone
+from enum import Enum
+from typing import List, Optional
+
+from fastapi import Depends, FastAPI, HTTPException, Response
+from pydantic import BaseModel, Extra, Field
+from requests import HTTPError
+from uvicorn import Config, Server
+
+from boefjes.clients.bytes_client import BytesAPIClient
+from boefjes.clients.scheduler_client import SchedulerAPIClient, TaskStatus
+from boefjes.config import settings
+from boefjes.job_handler import (
+    _collect_default_mime_types,
+    _find_ooi_in_past,
+    get_environment_settings,
+    get_octopoes_api_connector,
+    serialize_ooi,
+)
+from boefjes.job_models import BoefjeMeta
+from boefjes.katalogus.local_repository import LocalPluginRepository, get_local_repository
+from octopoes.models import Reference
+
+app = FastAPI(title="Boefje API")
+logger = logging.getLogger(__name__)
+
+
+class UvicornServer(multiprocessing.Process):
+    def __init__(self, config: Config):
+        super().__init__()
+        self.server = Server(config=config)
+        self.config = config
+
+    def stop(self):
+        self.terminate()
+
+    def run(self, *args, **kwargs):
+        self.server.run()
+
+
+def run():
+    config = Config(app, host=settings.boefje_api_host, port=settings.boefje_api_port)
+    instance = UvicornServer(config=config)
+    instance.start()
+    return instance
+
+
+class BoefjeInput(BaseModel):
+    task_id: str
+    output_url: str
+    boefje_meta: BoefjeMeta
+
+    class Config:
+        extra = Extra.forbid
+
+
+class StatusEnum(str, Enum):
+    COMPLETED = "COMPLETED"
+    FAILED = "FAILED"
+
+
+class File(BaseModel):
+    name: Optional[str]
+    content: str = Field(..., contentEncoding="base64")
+    tags: Optional[List[str]]
+
+
+class BoefjeOutput(BaseModel):
+    status: StatusEnum
+    files: Optional[List[File]]
+
+
+def get_scheduler_client():
+    return SchedulerAPIClient(settings.scheduler_api)
+
+
+def get_bytes_client():
+    return BytesAPIClient(
+        settings.bytes_api,
+        username=settings.bytes_username,
+        password=settings.bytes_password,
+    )
```

Comment:
I don't like that we create a new object with every request. We can refactor to optimize this later. I think the best pattern for this would be defining the client objects as singleton in the files that define the classes.